### PR TITLE
basic transifex commands

### PR DESCRIPTION
**Pull request by sravfeyn**
@dannyroberts @twymer @esoergel take a look at this.

This is a initial crack at transifex scripts. The code doesn't feel like the way it should be and I have added few questions as comments. Do not merge this - I am putting this here to get your feedback and improve it.

### CODE REVIEW

File: fab/fabfile.py [hunk 1]
```diff
@@ -1225,3 +1225,50 @@ def selenium_test():
...
+    local("./manage.py makemessages -l {source_lang}".format(source_lang=SOURCE_LANGUAGE))
+    local("git add ./locale/{source_lang}/LC_MESSAGES/django.po".format(source_lang=SOURCE_LANGUAGE))
+    local("git commit -m 'Updated translation resource for source language'")
+
+
+@task
+def push_to_transifex():
+    build_new_translation_source()
+    # need to do some checks, so that we are not overriding transifex remote
+    local("tx push --source")
```
Comment (esoergel) [inline fab/fabfile.py]: Will pushing override changes and not just update?  If that's the case, we should pull first, then `makemessages` and push.

File: fab/fabfile.py [hunk 2]
```diff
@@ -1225,3 +1225,50 @@ def selenium_test():
...
+@task
+def fetch_from_transifex(lang):
+    local("tx pull --language={language} -f".format(language=lang))
+
+
+@task
+def fetch_and_commit_translations_for_all_languages():
+    project_root = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+    sys.path.append(project_root)
+    import settings
```
Comment (sravfeyn) [inline fab/fabfile.py]: Is using django settings this way okay? I don't see django settings being imported in whole of fab at all, any reason?

Comment (czue) [inline fab/fabfile.py]: fabric is run outside of a django environment so generally it's not recommended that you try and do anything with settings.

typically the way we work around this is to have a management command in the django environment that fabric calls. you can search this file for `manage.py` for lots of examples

Comment (sravfeyn) [inline fab/fabfile.py]: Cool, will tunnel this logic through management-command.

File: fab/fabfile.py (hunk 1 above)
Comment (sravfeyn) [inline fab/fabfile.py]: Good call. I wanted to pull all once and commit before making and pushing new translations. Will add that logic.

**Comment by dannyroberts** [comment]
@sravfeyn Hey I'd strongly recommend doing this as a standalone script for a number of reasons—one of which is just that this isn't really what fab is for. If you want to get nice syntax for working with the shell from python, you can us `sh`. For example,

```python
local("./manage.py compilemessages")
local("git add locale")
local("git commit -m 'updated translations for target languages'")
```

becomes

```python
sh.python('manage.py', 'compilemessages')
sh.git.add('locale')
sh.git.commit(m="updated translations for target languages")
```

etc.

**Comment by proteusvacuum** [comment]
@sravfeyn do you still want to work on this? if not, please close this PR and delete the branch. Thanks!

**Comment by sravfeyn** [comment]
This was latter done by Ethan.

**Pull request closed without merging**